library/koji_tag_inheritance.py

```diff
@@ -180,16 +180,29 @@ def add_tag_inheritance(session, child_tag, parent_tag, priority, maxdepth,
     for rule in current_inheritance:
         if rule == new_rule:
             return result
-        if rule['priority'] == priority:
+        # if either name or priority has changed without the other, we need to
+        # delete then reinsert
+        if (rule['name'] == parent_tag) != (rule['priority'] == priority):
```

Comment:
I don't think this particular change to check `rule['name']` is necessary. In other words we do not specifically need to "delete" a pre-existing rule that matches the parent tag's name.

When we call the `setInheritanceData` RPC, the server will automatically overwrite any rules for the same parent name. This means I can update an existing parent's priority with a single "rule" to `setInheritanceData`.

Here's an example I tried on my dev server. When I have the following inheritance set for my child tag:

```
$ kojidev taginfo inheritance-1-child
Tag: inheritance-1-child [8]
Arches: None
Groups: 
Tag options:
Inheritance:
  10   .... inheritance-1-parent-a [9]
```

I can update the "parent-a" inheritance relationship priority from "10" to "20: with the following:

```python
rules = [
 {'child_id': 8,
  'intransitive': False,
  'maxdepth': None,
  'noconfig': False,
  'parent_id': 9,
  'pkg_filter': '',
  'priority': 20},
]

session.setInheritanceData('inheritance-1-child', rules)
```

We only need to delete existing rules that have the same priority integer. The hub does indeed raise when I try to set  a new parent tag to priority "10" and I do not set "delete link" on the old one that is priority "10":

```
__koji__kojidev.GenericError: Inheritance priorities must be unique (pri 20: [9, 10] )
```

Maybe we should document this behavior in the setInheritanceData docstring in Koji Hub.

Reply:
It's needed for the case where you have two parents, e.g. `parent-1` and `parent-2`, and you want to add a rule that conflicts with the name of one and the priority of the other. I will try to create a reproducing test case later.